index.js

```diff
@@ -5,65 +5,100 @@ var Alexa = require('alexa-app');
+var prompt = 'With CARVIS you can order the cheapest or fastest car available. For example, you can say, CARVIS, find me the cheapest ride to Hack Reactor';
+var reprompt = 'Tell me to book the cheapest or fastest car, and where you want to go';
 
-  res.say(prompt).reprompt(prompt).shouldEndSession(false);
+app.launch(function(req, res) {
+  res.say(prompt).reprompt(reprompt).shouldEndSession(false);
 });
 
 app.intent('GetEstimate', {
   'slots': {
     'MODE': 'MODE',
-    'DESTINATION': 'AMAZON.LITERAL'
+    'DESTINATION': 'DESTINATION',
+    'DESTINATION_ONE': 'DESTINATION_ONE',
   },
-  // TODO: configure these properly. also look into ./resources/*
+  'utterances': [
+    '{Find|Get|Order|Call|Book} {a|one|the|me the|me a} {MODE} {car|ride} to {DESTINATION}'
+  ]
 },
   function(req, res) {
-    var userId = request.userId; // the unique alexa session userId
-    console.log('userId:', userId);
-
+    var userId = req.userId; // the unique alexa session userId
     var mode = req.slot('MODE'); // cheapest or fastest
-    var reprompt = 'Tell me to book the cheapest or fastest car, and where you want to go';
+    
+    // find the DESTINATION slot that is populated in this request
+    console.log('req.data.request.intent.slots', req.data.request.intent.slots);
+    var destination = _.filter(req.data.request.intent.slots, function(slotValue, slotKey) {
+      return (slotValue.value && slotValue.value.length > 0 && slotKey.includes('DESTINATION'));
+    })[0].value;
+    console.log('Alexa thinks my destination is', destination);
+
     // todo: grab user location?
-    if (_.isEmpty(mode)) {
-      var prompt = 'I didn\'t catch that. Please try again';
+    if (_.isEmpty(mode) || _.isEmpty(destination)) {
+      prompt = 'I didn\'t catch that. Please try again';
       res.say(prompt).reprompt(reprompt).shouldEndSession(false);
     } else {
-      // TODO: where does location come in?
-      // TODO: rideHelper.placesCall(place); -- once for pickup & destination?
-      // TODO: call rideHelper.getEstimate(mode);
+      rideHelper.placesCall(destination, function(placeDesc, destCoords) {
+        rideHelper.getEstimate(mode, destCoords, function(winner) {
+          var answer = formatAnswer(winner, mode, placeDesc);
+          res.say(answer).send();
+        });
+      });
 
-      // we need to include this to end session
-      res.say(prompt).reprompt(reprompt).shouldEndSession(true);
     }
+    // Intent handlers that don't return an immediate response (because they 
+    // do some asynchronous operation) must return false
+    return false;
   }
 );
 
-// need to be included to pass Amazon reviews.
+// need to be included to pass Amazon reviews
 app.intent('AMAZON.StopIntent', exitFunction);
 app.intent('AMAZON.CancelIntent', exitFunction);
 app.intent('AMAZON.HelpIntent', helpFunction);
 
+var formatAnswer = function(winner, mode, placeDesc) {
+  mode = mode.includes('cheap') ? 'cheapest' : 'fastest';
```

Comment:
ternary is a bit confusing - it substitutes 'cheapest' into the parens? also we don't consider `fast` ?